refactor to remove default discovery in all dmsg clients

therealssj opened the pull request:
fixes: #20

jdknives commented on README.md:
```diff
@@ -20,11 +20,16 @@ serverPort := uint16(8080)
 
 // prepare the server
 sPK, sSK := cipher.GenerateKeyPair()
+dmsgC := &DMSGClient{
```
The capitalization of dmsg is confusing for me. Either we capitalize or do not. But that should be consistent. Both in comments and in code.

sn-srdjan replied:
It was initially done this way to mimic how Go handled [HTTP](https://golang.org/pkg/net/http/#HandlerFunc.ServeHTTP) and/or [TCP](https://golang.org/pkg/net/#DialTCP) (examples linked to each protocol).

Of course, I'm fine with either option, just giving some background on this...

evanlinjin replied:
In dmsg repo, we use the lowercase spelling. So this should be used.

jdknives commented on client-instance.go:
```diff
@@ -10,28 +10,50 @@ import (
-	once      sync.Once
+// Defaults for dmsg configuration, such as discovery URL
+const (
+	DefaultDiscoveryURL = "http://dmsg.discovery.skywire.cc"
```
I would not make the test deployment the default. For now it makes more sense, but as soon as Murphy updates prod, we should use prod.

therealssj replied:
whats the prod url?

jdknives replied:
"http://dmsg.discovery.skywire.skycoin.com"

jdknives commented on client-instance.go:
```diff
@@ -10,28 +10,50 @@ import (
 
-	errWrongPubKeyUsed error = errors.New("dmsg client already initialized with different pub key")
-	errWrongSecKeyUsed error = errors.New("dmsg client already initialized with different sec key")
+	errCreate error = errors.New("dmsg client don't exists and was not created successfully")
```
"Does not exist"

jdknives commented:
Would make sense imo to also run travis checks.

evanlinjin commented:
This is not your problem, but just noting some problems with the code base.
1. `dmsg-http` should really be `dmsghttp`.
2. `dmsghttp` should NOT be instantiating it's own `*dmsg.Client` instance. Instead, it should ask the user to provide a `*dmsg.Client`.

evanlinjin commented:
Oh and if you can run `go mod tidy` to clean up the the imports, that will be great.

therealssj commented:
made changes according to the comments @evanlinjin @jdknives

sn-srdjan commented on dmsg-http.go:
```diff
@@ -4,29 +4,24 @@ import (
+func Client(dmsgC *dmsg.Client) *http.Client {
+	transport := Transport{
+		DmsgClient: dmsgC,
 		RetryCount: 20,
```
Since refactor is in progress RetryCount should probably be moved next to clientTimeout

evanlinjin replied:
I’m not sure if RetryCount is even needed. I feel like it can be removed.

Chris-PS replied:
It was a requirement from Brandon when we started, that's why it's there

sn-srdjan commented on dmsg-http_test.go:
```diff
@@ -19,7 +20,7 @@ const (
 )
 
-func TestDMSGClient(t *testing.T) {
+func TestDmsgHTTP(t *testing.T) {
```
Guess this should be either TestDmsgHttp or TestHTTPOverDMSG / TestHTTPUsingDMSG

therealssj replied:
I looked at the naming in the standard library and it was HTTP in that

evanlinjin commented on server.go:
```diff
@@ -2,44 +2,28 @@ package dmsghttp
-// Default dicovery URL is stored as dmsghttp.DefaultDiscoveryURL
+// Data includes parameters to instantiate a dmsgclient and a port on which server will listen.
 type Server struct {
-	PubKey    cipher.PubKey
```
I don’t think the dmsghttp.Server type is needed. It’s not any more complicated just to use http.Server directly. Keeping this may confuse people as this is similarly named to http.Server but has a different interface.

therealssj replied:
we also dont need the client then 
all we need is the transport

therealssj commented:
here's a list of changes:

- remove dmsghttp client and dmsghttp server
- remove retry from dmsg transport 
- added travis
- added a Makefile
- cleaned up the imports (also dmsg now uses the develop branch)
- Updated tests and README for the above changes

removed the retry because its not difficult for the end user to just do a retry themselves
letting the user handle the error of dialing to the dmsg client before retrying is better than letting the transport handle it 

also there are libs like: https://github.com/hashicorp/go-retryablehttp
which can be used as drop in replacement for the default http client which have much better functionalities for retrying

sn-srdjan commented on transport.go:
```diff
@@ -4,70 +4,51 @@ import (
-		break
-	}
+
+	stream, streamErr = t.DmsgClient.DialStream(context.Background(), serverAddress)
```
Regarding this retry count, I'm sure this Dial was breaking back with a dmsg milestone1 implementation when we have multiple requests coming in on CXO (that is using dmsghttp to communicate). 
Sometimes it successfully dialed only after more then 10 retries when server is busy. Occasionally  dmsg server was not even under heavy load when this dial was breaking, but again that was with different implementation present in milestone1.

Maybe it would be reasonable to add a test case with a couple of parallel request to make sure each of them is served without this RetryCount gimmick.
